hack/go-lint.sh

```diff
@@ -16,14 +16,8 @@
 #     You should have received a copy of the GNU General Public License
 #     along with Nexus Operator.  If not, see <https://www.gnu.org/licenses/>.
 
+# it's expected to have GOPATH in your PATH
+# see this link to understand why we are using a different go.mod file: https://github.com/golang/go/issues/34506
+which golangci-lint >/dev/null || go get -modfile=go.tools.mod github.com/golangci/golangci-lint/cmd/golangci-lint@v1.27.0
```

Comment:
This is pretty cool! I didn't know we could do that with `-modfile`.

Why are we specifying the version here though? I mean, we're already specifying the dependency and its version on `go.tools.mod`, so keeping it here as well would result in needing to update both `go.tools.mod` and this command whenever we bump versions.

It could also lead to this command overwriting `go.tools.mod` if we forget to change it here.

```suggestion
 which golangci-lint >/dev/null || go get -modfile=go.tools.mod github.com/golangci/golangci-lint/cmd/golangci-lint
```

Wdyt?

Reply:
We must add the version, otherwise `go get` will take the last commit from that repo, which is not something that we want.

I agree that we should have a way to automate this version update, but won't cause any problems not updating it. The only thing is that we have to update the CI and the local script every time.

Reply:
Oh right, it downloads the latest release. I wasn't testing it correctly when I proposed the change. I suppose we need to keep it this way then.